tests/rptest/clients/kubectl.py

```diff
@@ -17,8 +17,31 @@
 
 
 def is_redpanda_pod(pod_obj: dict[str, Any], cluster_id: str) -> bool:
-    """Returns true if the pod API object name matches the Redpanda pattern."""
-    return pod_obj['metadata']['name'].startswith(f'rp-{cluster_id}')
+    """Returns true if the pod looks like a Redpanda broker pod"""
+
+    # Azure behaves this way.  This is also the 'new' way going forward (circa 2024/7)
+    # We look for pods whose metadata indicates that it is part of a statefulset, and that
+    # the pod names are generated with the template "redpanda-broker-...".
+    # False positives are quite unlikely with this criterion:
+    # Scenario would be:
+    # - Another Statefulset
+    # - That reuses the SAME generateName (bad!)
+    try:
+        if pod_obj['metadata']['generateName'] == 'redpanda-broker-':
+            if pod_obj['metadata']["labels"][
+                    "app.kubernetes.io/component"] == "redpanda-statefulset":
+                return True
+    except KeyError:
+        pass
+
+    # Other providers like AWS / GCP behave this way ("the old way")
+    try:
```

Comment:
Why do we put the try/except here in the second (original) case? In the old method we just threw. For our extra effort it just seems this would mask some errors.

Reply:
The intention is definitely to leave old behavior alone while also enabling Azure to function. Reverting the fallback code.